lib/data_source/client/github.rb

```diff
@@ -29,30 +42,86 @@ def user_repos
           direction: 'desc',
           per_page: 100
         }
-        with_cache('user_repos') do
-          request('/user/repos', params)
+        responses = with_cache(:user_repos) do
+          all_user_repos = merge_multipage_results('/user/repos', params, 100)
+          all_user_repos
+        end
+
+        all_user_repos = Array.new
+        responses.each do |response|
+          all_user_repos = all_user_repos + response
         end
+        all_user_repos
       end
 
       def user_pulls
-        modifiers = org_modifiers('is:pr', 'user:@me', 'state:open', 'involves:@me')
+        if @pr_all_involve_me
+          modifiers = ['is:pr', 'state:open', "involves:#@me_account"]
+        else
+          modifiers = org_modifiers('is:pr', "user:#@me_account", 'state:open', "involves:#@me_account")
+        end
         params = search_params('', modifiers).merge(
           per_page: 100
         )
-        response = with_cache('user_pulls') do
-          request('/search/issues', params)
+        responses = with_cache(:user_pulls) do
+          all_user_pulls = merge_multipage_results('/search/issues', params, 100)
+          all_user_pulls
+        end
+        all_user_pulls = Array.new
+        responses.each do |response|
+          all_user_pulls = all_user_pulls + response[:items]
         end
-        response[:items]
+        all_user_pulls
       end
 
       private
 
+      def merge_multipage_results(path, params, per_page)
+        params[:per_page] = per_page
+        params[:page] = 1
+        res = request(path, params)
+        raise res[:message] unless res.is_a?(Net::HTTPSuccess)
+
+        result = Array.new
+        result.append deserialize_body(res.body)
+
+
+        page_count = 1
+        if res.key?("Link")
+          res["Link"].split(",").map do |result|
+            page_num, rel = result.match(/&page=(\d+)>; .*?"(\w+)"/i).captures
+            page_count = page_num.to_i if rel == "last"
+          end
+        end
+
+        (2..page_count).step(1) do |n|
+          params[:page] = n
+          params[:per_page] = per_page
+          part_res = request(path, params)
+          if not part_res.is_a?(Net::HTTPSuccess)
+            result = nil
+            break
+          else
+            result.append deserialize_body(part_res.body)
+          end
+        end
+        result
+      end
+
+
       def search_params(query, modifiers)
         { q: "#{query} #{modifiers.join(' ')}" }
       end
 
       def org_modifiers(*initial)
-        orgs = with_cache('user_orgs') { request('/user/orgs') }
+        orgs = with_cache(:user_orgs) do
+          res = request('/user/orgs')
+          if not res.is_a?(Net::HTTPSuccess)
+            nil
+          else
+            deserialize_body(res.body)
+          end
```

Comment:
We should return an empty array if response is not successful.

```suggestion
          res.is_a?(Net::HTTPSuccess) ? deserialize_body(res.body) : []
```